onnx/defs/schema.cc

```diff
@@ -724,19 +742,33 @@ OpSchema& OpSchema::FillUsing(const std::function<void(OpSchema&)>& populator) {
   return *this;
 }
 
-void OpSchema::BuildFunction() {
-  function_body_.set_name(this->name_);
-  function_body_.set_doc_string(this->doc_);
-  function_body_.set_since_version(this->since_version_);
-  function_body_.set_status(OperatorStatus(1 - (int)this->support_));
+// only add input to the function body if it is used by a body's node.
+bool FilterFunctionBodyInputByName(const std::string& input_name, const FunctionProto& function_body)
+{
+  for (size_t i = 0; i< function_body.node_size(); i++)
+  {
+    auto node = function_body.node((int)i);
+    for (size_t j = 0; j < node.input_size(); j++)
+      if (node.input((int)j) == input_name)
+        return true;
+  }
+  return false;
+}
+
+void OpSchema::BuildFunction(FunctionProto& function_body) {
+  function_body.set_name(this->name_);
+  function_body.set_doc_string(this->doc_);
+  function_body.set_since_version(this->since_version_);
+  function_body.set_status(OperatorStatus(1 - (int)this->support_));
   for (auto& i : inputs_) {
-    function_body_.add_input(i.GetName());
+    if (FilterFunctionBodyInputByName(i.GetName(), function_body))
```

Comment:
I am not sure if this filtering is a good idea, because the input signature of the function should match parameters passed at call-site. This filtering could break things. It seems okay if a function body has an unused input.

Reply:
discussed with @gramalingam offline. The op signature need to be kept otherwise the provider kernel will fail. I will update this according to reviewers' comment.